pennylane/circuit_drawer/draw_mpl.py

```diff
@@ -253,6 +253,8 @@ def draw_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, **kwarg
                 control_wires = [wire_map[w] for w in op.control_wires]
                 target_wires = [wire_map[w] for w in op.wires if w not in op.control_wires]
 
+                extra_width = -1 if op.name == "Barrier" else 0
```

Comment:
I'd probably recommend using `special_cases` for the Barrier. 

 For now, I would just do nothing for Barriers, `ops.Barrier: lambda *args: pass`, leaving the space open for user annotations and modifications.  

If there's a need, later I could add a special method to `qml.circuit_drawer.MPLDrawer` and then update the special function that draws a Barrier.

Reply:
> I played around with the code a bit locally. With the new beta QNode, there is a separation between gradient expansion and device expansion. The beta QNode will store the pre-device expansion version of the tape.
> 
> If we always return `[]` from `decomposition`, and remove support for the operation from the device, then we have can `Barrier` always remain in `beta_qnode.qtape` without it being in the actually executed tape.
> 
> @josh146 How long till the beta QNode is made the default? That will impact how we decide to handle `Barrier`.
> 
> I edited `decomposition` to only return `[]`, and this works.
> 
> ```
> dev = qml.device('lightning.qubit', wires=2)
> 
> @qml.beta.qnode(dev)
> def circuit():
>     qml.PauliX(0)
>     qml.Barrier(wires=(0,1))
>     qml.PauliX(1)
>     return qml.state()
> 
> circuit()
> 
> >>>print(circuit.qtape.operations)
> [PauliX(wires=[0]), Barrier(wires=[0, 1]), PauliX(wires=[1])]
> ```

As for this, in the compiler it failed as it removed gates when it should not have. That is why I have not modified it